internal/transport/defaults.go

```diff
@@ -47,3 +47,8 @@ const (
 	defaultClientMaxHeaderListSize = uint32(16 << 20)
 	defaultServerMaxHeaderListSize = uint32(16 << 20)
 )
+
+// MaxStreamIDForTesting is the upper bound for the stream ID before the current
+// transport gracefully closes and a new transport is created for subsequent RPCs.
+// This is set to 75% of math.MaxUint32. It's exported so that tests can override it.
+var MaxStreamIDForTesting = uint32(float32(math.MaxUint32) * 0.75)
```

Comment:
Also a nit: I'd simplify this to something like `uint32(3*1024*1024*1024)` or `3_221_225_472` or `0xc0000000`.  (Basically, extra casts and floats (in general) should be avoided.)

Reply:
i missed this one.. updating this now